test/appium/views/home_view.py

```diff
@@ -267,63 +270,68 @@ def get_chat_from_activity_center_view(self, chat_name):
         chat_element = ActivityCenterElement(self.driver, chat_name[:25])
         return chat_element
 
-    def handle_contact_request(self, username: str, accept=True):
+    def handle_contact_request(self, username: str, action='accept'):
         if self.notifications_unread_badge.is_element_displayed(30):
             self.open_activity_center_button.click()
         chat_element = ActivityCenterElement(self.driver, username[:25])
-        if accept:
-            self.driver.info("Accepting contact request for %s" % username)
+        if action == 'accept':
+            self.driver.info("Accepting incoming CR for %s" % username)
             chat_element.accept_contact_request()
-        else:
-            self.driver.info("Rejecting contact request for %s" % username)
+        elif action == 'decline':
+            self.driver.info("Rejecting incoming CR for %s" % username)
             chat_element.decline_contact_request()
+        elif action == 'cancel':
+            self.driver.info("Canceling outgoing CR for %s" % username)
+            chat_element.cancel_contact_request()
+        else:
+            self.driver.fail("Illegal option for CR!")
         self.close_activity_centre.click()
         self.chats_tab.wait_for_visibility_of_element()
 
     def get_username_below_start_new_chat_button(self, username_part):
         return Text(self.driver,
                     xpath="//*[@content-desc='enter-contact-code-input']/../..//*[starts-with(@text,'%s')]" % username_part)
 
-    def add_contact(self, public_key, add_in_contacts=True, nickname=''):
-        self.driver.info("## Starting 1-1 chat, add in contacts:%s" % str(add_in_contacts), device=False)
-        self.plus_button.click_until_presence_of_element(self.start_new_chat_button)
-        chat = self.start_new_chat_button.click()
+    def add_contact(self, public_key, nickname='', remove_from_contacts=False):
+        self.driver.info("Adding user to Contacts via chats > add new contact")
+        self.new_chat_button.click_until_presence_of_element(self.add_a_contact_chat_bottom_sheet_button)
+        self.add_a_contact_chat_bottom_sheet_button.click()
+
+        chat = self.get_chat_view()
         chat.public_key_edit_box.click()
         chat.public_key_edit_box.send_keys(public_key)
-        one_to_one_chat = self.get_chat_view()
-        chat.confirm()
-        # chat.confirm_until_presence_of_element(one_to_one_chat.chat_message_input)
-        if add_in_contacts and one_to_one_chat.add_to_contacts.is_element_displayed():
-            one_to_one_chat.add_to_contacts.click()
+        chat.view_profile_new_contact_button.click_until_presence_of_element(chat.profile_block_contact)
+        if remove_from_contacts and chat.profile_remove_from_contacts.is_element_displayed():
+            chat.profile_remove_from_contacts.click()
```

Comment:
Why do wee need a logic related to rejecting a CR inside a method which is adding a contact? I'd suggest  to split it and  have  2 different methods like add_contact and reject_contact

Reply:
It is temp and I suggest to leave it as it is before introduction of new interface of contact's profile, which is completely different

Reply:
that makes sense